src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -848,5 +848,66 @@ private static bool SupportsFoldingRange(JToken token)
 
             return result?.Response;
         }
+
+        public override async Task<JToken?> ProvideCompletionsAsync(
```

Comment:
If `JToken?` the most specific we can be? I feel like surely we have more guarantees than that, even if we have some JToken's under the covers.

Reply:
Or did you do this to avoid converting just to convert back to JToken? If that's the case probably comment it.

Reply:
ah ya I totally forgot to mention this. It's the pattern we should start utilizing. Basssssically I've heard that things eventually make their way into JTokens at various portions of the stack so converting away from them doesn't make much of a difference. Therefore, in order to avoid one extra conversion I was leaving as JToken (as you rightfully guessed). Can leave comment